File: packages/types/src/events/conversation-events.ts
/**
 * Conversation-related event type definitions
 */

import type { ID } from "../common.js";
import type { BaseEvent } from "./base.js";

/**
 * Message Add Event Type
 */
export interface MessageAddedEvent extends BaseEvent {
  type: "MESSAGE_ADDED";
  /** Node ID */
  nodeId?: ID;
  /** message role */
  role: string;
  /** Message content */
  content: string;
}

/**
 * Dialog state change event type
 */
export interface ConversationStateChangedEvent extends BaseEvent {
  type: "CONVERSATION_STATE_CHANGED";
  /** Node ID */
  nodeId?: ID;
  /** Number of messages */
  messageCount: number;
  /** Token usage */
  tokenUsage: number;
}

File: crates/wf-types/src/events/conversation_events.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct MessageAddedEvent {
    pub base: super::BaseEvent,
    pub node_id: Option<super::super::Id>,
    pub role: String,
    pub content: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct ConversationStateChangedEvent {
    pub base: super::BaseEvent,
    pub node_id: Option<super::super::Id>,
    pub message_count: u32,
    pub token_usage: u32,
}
